src/ui/itemTypes.js

```diff
@@ -0,0 +1,148 @@
+import { ITEMS } from "../constants/items.js";
+import { MATERIALS } from "../constants/materials.js";
+
+export function getItemClass(itemName, material, scene) {
+    let itemData = ITEMS[itemName];
+    let matData = MATERIALS[itemData.material][material];
+
+    switch (itemName) {
+        case "Bones":
+            return new Bones(itemData, matData, scene);
+            break;
+        case "Logs":
+            return new Logs(itemData, matData, scene);
+            break;
+    }
+}
```

Comment:
I'm not sure how to do this with how the weapons are currently organized. In items.js, "Bones" and "Logs" are currently sitting directly under the parent ITEM whereas since there are multiple types of weapons, the data for other weapons is under ITEM.Weapon.Dagger for example. 

A couple ways for this to work, 
1) Case statement has all weapon names there and creates a Weapon class if it is listed
2) One of the input parameters to the function is the explicit path to the data, for example "Weapon.Scimitar" or "Weapon.Crossbow"

Reply:
I'm not sure we should generalize this. Each "entity" should be a class that will create/drop/instantiate other specific item objects directly. We should just call their constructors directly within those classes. I'm also more in favor of keeping classes to their own files.

Reply:
So you think there should be a "Dagger" class rather than using a "Weapon" or "MeleeWeapon" class for it? I feel like that would generate a lot of duplicate code

Reply:
The better the class hierarchy the less amount of duplicate code there will be. There should be classes for each of these: weapon -> meleeweapon -> dagger -> bronze dagger. Allowing the leaf classes (bronze dagger here) to use any of its parents methods/attributes will minimize duplicate code. Drop/Examine can be defined much higher than other potential options. Examine can just pull from an examine text attribute that is required.

Reply:
The methods that a bronze dagger would need are the same as a mithril dagger would, the only differences between them are data like damage and material name. These would be stored for all weapons so I don't see the benefit of adding a new class for each individual item.

Reply:
If for no other reason, do it for organization. It makes more sense to have actual items one will see in-game be the leaf classes than to have a "logical parent" be the leaf.

Reply:
Alright, I have an idea for how this could be cleanly implemented but it would require all the leaf classes to be in one file:
Item-classes.js
```
export class BronzeDagger() {}
export class IronDagger() {}
...
```
Item-types.js
```
import * as ItemClasses from "./item-classes.js";

export function getItemClass(itemName, material, scene) {
    return new ItemClasses[material + itemName]();
}
```
I've tried similar things for classes and I'm pretty sure this will work and remove the need for a massive case statement. This doesn't allow for separate files for each class, but only allows for a separate file for the parent Item class vs all the children. What do you think?

Actually you might be able to do:
```
import BronzeDagger as ItemClasses.BronzeDagger...
import IronDagger as ItemClasses.IronDagger...
...
```

In which case you have the benefits of both, but a bigger header instead.

Reply:
Can you store the item classes within the drop tables? That would remove the need for getItemClass

Reply:
I think you can but there would still need to be some method to recreate items just based on text when you refresh from cookies. When you convert classes to JSON they lose their functions

Reply:
Could you store the constructor?

Reply:
I'm not sure. Do you have any idea of what that would look like?

Reply:
Idk JS well enough, but if you cant directly story the constructor method then you might be able to use a static factory method to construct the objects

Reply:
After looking into it, it's possible but the work we'd have to do for it and the amount of text data saved doesn't make it an attractive option.

I did find out that you can dynamically import which would reduce the need for all these lines of imports so it wouldn't be that bad.

bronze-dagger.js
```
export default class BronzeDagger() {}
```

item.js
```
export function getItemClass(item, material) {
    let item = await import("./items/" material +"-" + item + ".js"); // "./items/iron-dagger.js"
    return item.default();
}
```

Reply:
I'll look into factory methods in js though

Reply:
I decided to create a file that I called item-manifest that has the names and file paths for both the classes and the images. This is used in the getItemClass function to dynamically import the class. This also should speed up load times on startup once all the weapons have been implemented because it won't have to loop through as many objects to find the item/material pairings, it'll just run down this one list.